internal/policy/verify.go

```diff
@@ -599,22 +716,26 @@ func getCommits(repo *gitinterface.Repository, entry *rsl.ReferenceEntry) ([]git
 	return repo.GetCommitsBetweenRange(entry.TargetID, priorRefEntry.TargetID)
 }
 
-func verifyGitObjectAndAttestations(ctx context.Context, policy *State, target string, gitID gitinterface.Hash, authorizationAttestation *sslibdsse.Envelope, approverKeyIDs *set.Set[string]) (string, error) {
+func verifyGitObjectAndAttestations(ctx context.Context, policy *State, target string, gitID gitinterface.Hash, authorizationAttestation *sslibdsse.Envelope, approverKeyIDs *set.Set[string], verifyMergeable bool) (string, bool, error) {
 	verifiers, err := policy.FindVerifiersForPath(target)
 	if err != nil {
-		return "", err
+		return "", false, err
 	}
 
-	return verifyGitObjectAndAttestationsUsingVerifiers(ctx, verifiers, gitID, authorizationAttestation, approverKeyIDs)
+	return verifyGitObjectAndAttestationsUsingVerifiers(ctx, verifiers, gitID, authorizationAttestation, approverKeyIDs, verifyMergeable)
 }
 
-func verifyGitObjectAndAttestationsUsingVerifiers(ctx context.Context, verifiers []*Verifier, gitID gitinterface.Hash, authorizationAttestation *sslibdsse.Envelope, approverKeyIDs *set.Set[string]) (string, error) {
+func verifyGitObjectAndAttestationsUsingVerifiers(ctx context.Context, verifiers []*Verifier, gitID gitinterface.Hash, authorizationAttestation *sslibdsse.Envelope, approverKeyIDs *set.Set[string], verifyMergeable bool) (string, bool, error) {
 	if len(verifiers) == 0 {
 		// This target is not protected by gittuf policy
-		return "", nil
+		return "", false, nil
 	}
 
-	verifiedUsing := ""
+	var (
+		verifiedUsing                       string
+		rslEntrySignatureNeededForThreshold bool
```

Comment:
> @wlynch: What is this tracking and why is it only checked the 2nd time?

This is tracking if we're already meeting threshold with approval attestations or if one final approval is needed via the signature on the RSL entry. Consider a rule that has threshold = 2. This can be met in two ways:
a) signature 1 on the RSL entry for the ref and signature 2 on an attestation
b) both signatures on the attestation

In the VerifyMergeable case, we may have all the approvals needed already, meaning the signature on the RSL entry does not matter. OTOH, something may still be good to merge _as long as the merge is by an authorized user who signs the RSL entry_. So, we track this and set it to true only if threshold-1 (rather than threshold) is met.

Reply:
This is captured in the docstring inline.